analyses/long-format-table-utils/annotator/download-annotation-data.R

```diff
@@ -0,0 +1,205 @@
+  if (!is.character(gn_vec)) {
+    stop(paste0("mygene query returns non-character refseq.protein.\n",
+                "Check query results. Revise download-annotation-data.R ",
+                "to handle non-character values."))
+  }
+  # remove NAs
+  uniq_rm_na_gn_vec <- sort(unique(purrr::discard(gn_vec, is.na)))
+
+  clp_gn_str <- paste(uniq_rm_na_gn_vec, collapse = ",")
+  if (identical(clp_gn_str, "")) {
+    # no non-NA name for the gene query
+    # dplyr::summarise needs function return value to be character for a
+    # character column. as.character(NA) is still NA rather than "NA".
+    return(as.character(NA))
+  } else {
+    return(clp_gn_str)
+  }
+}
+# # test cases
+# collapse_name_vec(c("a", "b"))
+# collapse_name_vec(c("a", "b", NA))
+# collapse_name_vec(c("a", "a", "a", "b", "d"))
+# collapse_name_vec(c("d", "a", "a", "b", "d"))
+# collapse_name_vec(c(NA, character(0)))
+# is.na(collapse_name_vec(c(NA, character(0))))
+# class(collapse_name_vec(c(NA, character(0))))
+# collapse_name_vec(c(NA, NA, NA, character(0)))
+# collapse_name_vec(character(0))
+# # folloing cases should fail
+# collapse_name_vec(c(1))
+
+
+
+
+# Set up directory paths -------------------------------------------------------
+# Adapted from the oncoprint-landscape module.
+#
+# Detect the ".git" folder -- this will in the project root directory. Use this
+# as the root directory to ensure proper execution, no matter where it is called
+# from.
+#
+# This only works if the working directory is OpenPedCan-analysis or a
+# subdirectory of OpenPedCan-analysis
+#
+# root_dir is the absolute path of OpenPedCan-analysis
+tryCatch(
+  {
+    root_dir <- rprojroot::find_root(rprojroot::has_file(".git/index"))
+  },
+  error = function(err_cond) {
+    # adapted from http://adv-r.had.co.nz/Exceptions-Debugging.html
+    err_cond$message <- paste0(
+      err_cond$message,
+      "\nTry re-running this script with working directory as ",
+      "OpenPedCan-analysis or a subdirectory of OpenPedCan-analysis.\n")
+    stop(err_cond)
+  }
+)
+
+input_data_dir <- file.path(root_dir, "data")
+
+output_data_dir <- file.path(
+  root_dir, "analyses", "long-format-table-utils", "annotator",
+  "annotation-data")
+
+if (!dir.exists(output_data_dir)) {
+  dir.create(output_data_dir)
+}
+
+
+
+# Read input data --------------------------------------------------------------
+# ensg hugo rmtl mappings
+ensg_hugo_rmtl_df <- dplyr::distinct(
+  readr::read_tsv(file.path(input_data_dir, "ensg-hugo-rmtl-v1-mapping.tsv"),
+                  col_types = readr::cols(.default = readr::col_guess())))
+
+# assert all ensg_ids and gene_symbols are not NA
+if (!identical(sum(is.na(ensg_hugo_rmtl_df$ensg_id)), as.integer(0))) {
+  stop(paste0("Found NA in ensg-hugo-rmtl-v1-mapping.tsv ensg_id.\n",
+              "Check if PedOT release data are downloaded properly.\n",
+              "If data is downloaded properly, submit a GitHub data question."))
```

Comment:
```suggestion
              "If data is downloaded properly, submit a GitHub data issue."))
```

Reply:
Thanks for the error message suggestions!